proposals/inline-classes.md

```diff
@@ -128,31 +213,79 @@ Since boxing doesn't have side effects as is, it's possible to reuse various opt
 ### Type mapping on JVM
 
 Depending on the underlying type of inline class and its declaration site, inline class type can be mapped to the underlying type or to the 
-wrapper type.  
-
+boxed type.  
 
 | Underlying Type \ Declaration Site | Not-null | Nullable | Generic |
 | --------------------------------------------- | --------------- | ---------------- | ---------------- |
-| **Reference type** | Underlying type | Underlying type | Wrapper type |
-| **Nullable reference type** | Underlying type  | Wrapper type | Wrapper type |
-| **Primitive type** | Underlying type  | Wrapper type | Wrapper type |
-| **Nullable primitive type** | Underlying type  | Wrapper type | Wrapper type |
+| **Reference type** | Underlying type | Underlying type | Boxed type |
+| **Nullable reference type** | Underlying type  | Boxed type | Boxed type |
+| **Primitive type** | Underlying type  | Boxed type | Boxed type |
+| **Nullable primitive type** | Underlying type  | Boxed type | Boxed type |
 
 For example, from this table it follows that nullable inline class which is based on reference type is mapped to the underlying type:
-```
+```kotlin
 inline class Name(val s: String)
 
 fun foo(n: Name?) { ... } // `Name?` here is mapped to String
 ```
 
+Also, if inline class type is used in generic position, then its boxed type will be used:
+```
+// Kotlin: sample.kt
+
+inline class Name(val s: String)
+
+fun generic(names: List<Name>) {} // generic signature will have `List<Name>` as for parameters type
+fun simple(name: Name) {}
+
+// Java
+class Test {
+    void test() {
+        String name = SampleKt.simple();
+        List<Name> ls = Samplekt.generic(); // from Java POV it's List<Name>, not List<String>
```

Comment:
In the Kotlin code `Name` is a parameter type.
In Java code `Name` is a return type.
`Name` is a parameter or a return type?

Reply:
Ah, indeed, `Name` and `List<Name>` should be used in return type position in Kotlin, thanks!